utils/network/0auth.ts

```diff
@@ -1,64 +1,96 @@
 import { AxiosRequestConfig } from 'axios';
-import oauth from 'axios-oauth-client';
-import tokenProvider from 'axios-token-interceptor';
-import { HttpServerError } from '#clients/exceptions';
+import { HttpServerError, HttpUnauthorizedError } from '#clients/exceptions';
 import constants from '#models/constants';
+import { logWarningInSentry } from '#utils/sentry';
 import {
   defaultAxiosInstanceFactory,
   cachedAxiosInstanceFactory,
   defaultCacheConfig,
 } from '.';
 
-export const httpClientOAuthGetFactory = (
-  token_url: string,
-  client_id: string | undefined,
-  client_secret: string | undefined
-) => {
-  if ((!client_id || !client_secret) && process.env.NODE_ENV === 'production') {
-    throw new HttpServerError('Client id or client secret is undefined');
+type IAccessToken = {
+  data: {
+    access_token: string;
+    expires_in: number;
+    scope: string;
+    token_type: 'Bearer' | string;
+  };
+  tokenExpiryTime: number;
+};
+
+export class httpClientOAuth {
+  private _token: IAccessToken | null;
+  private _cachedAxiosInstance;
+
+  constructor(
+    private token_url: string,
+    private client_id: string | undefined,
+    private client_secret: string | undefined
+  ) {
+    if (
+      (!this.client_id || !this.client_secret || !this.token_url) &&
+      process.env.NODE_ENV === 'production'
+    ) {
+      throw new HttpServerError('Client id or client secret is undefined');
+    }
+    this._cachedAxiosInstance = cachedAxiosInstanceFactory();
+    this._token = null;
   }
 
-  // function that get oauth2 token
-  const getAuthorizationCode = oauth.clientCredentials(
-    defaultAxiosInstanceFactory(constants.timeout.XS),
-    token_url,
-    client_id,
-    client_secret
-  );
+  refreshToken = async () => {
```

Comment:
refreshToken est en réalité getNewToken puisqu'on fait un get sur la route this.token_url